src/stampy_nlp/routes.py

```diff
@@ -111,10 +131,22 @@ def literature_api():
 
 
 @api.route('/extract', methods=['GET', 'POST'])
+@log_queries
 def extract_api():
     logger.debug('extract_api()')
     if request.method == "POST":
         query = request.form.query
     elif request.method == "GET":
         query = request.args.get("query", DEFAULT_QUERY)
     return jsonify(extract_qa(query))
+
+
+@api.route('/log_query', methods=['GET'])
+def log_query_endpoint():
```

Comment:
this can be called from the frontend to log tensorflow queries

Reply:
OK, but I'd rather not actually call or log *all* TF queries, since that'll be too much info. Maybe doesn't hurt to leave it as an option for now. I was thinking for the UI to periodically NLP endpoint to get the more "accurate" search results which will then automatically get logged with other queries.

Reply:
yup, totally agreed. But that would be up to the ui code